### DISCUSSION TITLE
Is there something special to namespace 'hello'?

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#316] [Q&A]
```
➜  crexx git:(master) ✗ rxc hello
Error in hello @ 6:5 - #IS_A_NAMESPACE, "hello"
```
This goes away if I rename the namespace to 'scratch'.

**Comment by adesutherland** [comment]
No ... it's a bug. Can you post the source please

**Comment by rvjansen** [comment]
```
options levelb
import rxfnsb
namespace hello


say "hello crexx!"
say 'hello crexx'
say hello crexx

/* some arithmetic */
say 1 + 1
say 0.2 - 0.1
say 800/81

say 'hello '||'Crexx'
return 0
```

**Reply by adesutherland** [reply]
I'll look at it. What's interesting is that line 8 should give an error ... a taken constant already defined as a namespace

**Reply by adesutherland** [reply]
When I run the above code I get

    Error in scratch @ 8:5 - #IS_A_NAMESPACE, "hello"

This is by design (the truth is I didn't think of this possibility). What is happening is that normally the hello on line 8 would be a taken constant but in this case the symbol is a namespace hence the error.

If you had

    options levelb
    namespace hello
    hello = 3

It would give the same error because hello is a namespace not a variable.

Does that make sense?

**Comment by rvjansen** [comment]
another funny thing: I try to compile the we.rexx outside of the development tree; gives me this:
```
➜  crexx git:(master) ✗ rxc we
Can't open input file .#hello.rxas
Error in we @ 7:9 - #FUNCTION_NOT_FOUND, "Date(\'b\')"
Error in we @ 8:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'23 Sep 1958\')"
Error in we @ 9:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'10 Mar 1962\')"
Error in we @ 10:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'23 Aug 1967\')"
Error in we @ 11:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'19 Sep 1992\')"
Error in we @ 12:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'26 Mar 1990\')"
Error in we @ 13:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'15 Apr 1934\')"
Error in we @ 14:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'6 Oct 1939\')"
Error in we @ 15:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'26 May 2017\')"
Error in we @ 16:25 - #FUNCTION_NOT_FOUND, "date"
Error in we @ 29:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'23 Sep 1958\')"
Error in we @ 30:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'19 Sep 1992\')"
Error in we @ 34:9 - #FUNCTION_NOT_FOUND, "date(\'n\',i,\'b\')"
Error in we @ 39:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'23 Sep 1958\')"
Error in we @ 40:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'26 Mar 1990\')"
Error in we @ 44:9 - #FUNCTION_NOT_FOUND, "date(\'n\',i,\'b\')"
Error in we @ 49:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'10 Mar 1962\')"
Error in we @ 50:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'19 Sep 1992\')"
Error in we @ 54:9 - #FUNCTION_NOT_FOUND, "date(\'n\',i,\'b\')"
Error in we @ 59:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'10 Mar 1962\')"
Error in we @ 60:12 - #FUNCTION_NOT_FOUND, "date(\'b\',\'26 Mar 1990\')"
Error in we @ 64:9 - #FUNCTION_NOT_FOUND, "date(\'n\',i,\'b\')"
Error in we @ 74:14 - #FUNCTION_NOT_FOUND, "date(\'b\',\'23 Aug 1967\')"
Error in we @ 75:41 - #FUNCTION_NOT_FOUND, "date(\'n\',twentythou,\'b\')"
24 error(s) in source file
```
note the ```Can't open input file .#hello.rxas```

```
➜  crexx git:(master) ✗ rxc -i /Users/rvjansen/apps/crexx_release/lib/rxfns/library we
[1]    76719 segmentation fault  rxc -i /Users/rvjansen/apps/crexx_release/lib/rxfns/library we
```
this is when I try to add the library; both attempts have the prototype for the Date() bif commented out.

So I am stuck for the moment, not understanding stuff enough at the moment.

**Comment by rvjansen** [comment]
.#hello.rxas was an alias (symbolic link) left behind by an aborted emacs session. I wonder why rxc tried to read it.

**Comment by rvjansen** [comment]
Ok, I have a bit more clarity now. The part that I missed until now:
1. on the rxc compile, there must be an ```-i /Users/rvjansen/apps/crexx_release/lib/rxfns```
2. on the rxvm, we must mention ```/Users/rvjansen/apps/crexx_release/lib/rxfns/library```

The difference is not entirely clear to me yet. Why are we not specifying the whole path on the -I to rxc? We mean the same library.rxbin. Or do we?

**Reply by adesutherland** [reply]
Sorry for the delay - stress at work! 

Yes because of slightly different use cases  

- "rxc -i" takes directories (; delimited) the compiler searches all files in those directories.
- rxvm has space delimited file names (without the rxbin extension) 

The truth is we need to make then consistent! I think by updating rxvm with an -i option just like rxc. Same for rxas

The reason that rxc has the more advanced logic is because of the use case where a developer has a bunch of project files (.rexx) in his project directory.
